### ISSUE TITLE
o_pp_tellu_pclean_AB.fits  

### ISSUE THREAD

**Issue by andrescarmona** [njcuk9999/apero-drs#651]
Hello Neil,

what command produces the o_pp_tellu_pclean_AB.fits  files?

i am running the reduction from scratch of an object using the distributed telluDB and calibDB

when it arrives at FTELLU1 

the is an error (example below)

the DRS looks for a "pre-cleaned" file in the /reduced/<night> directory

but as the reduction starts from scratch and its 
the first time the routine is called the file is not there…

the file that the DRS looks is however in the telluDB de Montreal and in the master_tellu_SPIROU.txt.

is that means because the file is in the telluDB it will look for it in the reduced folder?
instead of calculating as new?

Cheers and thanks!

Andres

extract of the error

```python
18:46:47.897- |FTELLU1[00277]|  ***************************************************************************
18:46:47.989-*|FTELLU1[00277]|I[40-001-00017]: Arguments used:
18:46:48.055-*|FTELLU1[00277]|
18:46:48.057-*|FTELLU1[00277]|	--DIRECTORY: /net/GSP/nas12c/big_spirou/FULL_REDUCTION_CALIBDB_06131/science_targets/HD189733/reduced/2018-07-29 
18:46:48.059-*|FTELLU1[00277]|	--FILES[0]: [EXT_E2DS_FF] 2294567o_pp_e2dsff_AB.fits 
18:46:48.060-*|FTELLU1[00277]|	--USE_TEMPLATE: True 
18:46:48.061-*|FTELLU1[00277]|	--DATABASE: True 
18:46:48.063-*|FTELLU1[00277]|	--PLOT: 0 
18:46:48.064-*|FTELLU1[00277]|	--DEBUG: None 
18:46:48.065-*|FTELLU1[00277]|	--PROGRAM: FTELLU1[00277] 
18:46:48.067-*|FTELLU1[00277]|	--IDEBUG: None 
18:46:48.068-*|FTELLU1[00277]|	--MASTER: None 
18:46:48.069-*|FTELLU1[00277]|	--BREAKPOINTS: None 
18:46:48.071-*|FTELLU1[00277]|	--BREAKFUNC: None 
18:46:48.072-*|FTELLU1[00277]|	--FORCE_INDIR: None 
18:46:48.073-*|FTELLU1[00277]|	--FORCE_OUTDIR: None
18:46:48.136- |FTELLU1[00277]| ***************************************************************************
18:46:53.890- |FTELLU1[00277]| ***************************************************************************
18:46:53.964- |FTELLU1[00277]|A[40-001-00020]: Processing file 1 of 1
18:46:54.039- |FTELLU1[00277]| ***************************************************************************
18:46:54.116-*|FTELLU1[00277]|I[40-019-00048]: File validated: OBJECT=HD189733 DPRTYPE=OBJ_FP
18:47:04.938- |FTELLU1[00277]|A[40-017-00036]: Wavelength solution read from filename.
18:47:04.940- |FTELLU1[00277]|	 Wave file = /net/GSP/nas12c/big_spirou/FULL_REDUCTION_CALIBDB_06131/science_targets/HD189733/calibDB/240041F6T7c_pp_e2dsff_AB_wavem_fp_AB.fits
18:47:05.487- |FTELLU1[00277]|A[40-017-00036]: Wavelength solution read from infile.
18:47:05.489- |FTELLU1[00277]|	 Wave file = 2294626c_pp_e2dsff_AB_wave_night_AB.fits
18:47:21.894- |FTELLU1[00277]|A[40-019-00043]: Reading pre-cleaned file from: /net/GSP/nas12c/big_spirou/FULL_REDUCTION_CALIBDB_06131/science_targets/HD189733/reduced/2018-07-29/2294567o_pp_tellu_pclean_AB.fits
18:47:21.962-!|FTELLU1[00277]|E[01-001-00012]: Read Error: File '2294567o_pp_tellu_pclean_AB.fits' could not be found in directory '/net/GSP/nas12c/big_spirou/FULL_REDUCTION_CALIBDB_06131/science_targets/HD189733/reduced/2018-07-29'. 
18:47:21.963-!|FTELLU1[00277]|	 function = io.drs_fits.py.readfits()
```

**Comment by njcuk9999** [comment]
I think I know what's going on here...  I'll try to explain:

The precleaned file is created by the mk_tellu and fit_tellu codes for each odometer code.For hot stars, as you know, we do mk_tellu, fit_tellu, mk_template, mk_telluand for science objects we do fit_tellu, mk_template, fit_telluAs there is repetition in doing mk_tellu and fit_tellu - we don't need to "preclean" a file three times (for hot stars) and two times (for science objects) so once we create the precleaned file for an odometer code we do not create it again.
We do this by saving the pre-cleaned file to the telluric database (you'll see them in there)... however it looks like I made a mistake and actually read the precleaned file from the reduced directory (which is fine in a "normal" reduction because the way the calib and tellu databases work is they first save to the reduced directory then are copied to the calib/tellu db if quality control pass.... 

So this is where your situation at LAM is different - you have the calib and tellu databases from Montreal - but your reduced directories do not have the precleaned files in - so it cannot find the equivalent precleaned file from the tellu database in the reduced directory. :( 

### Quick fix: 
There are a few options:

1. remove all entries in the master_tellu_DB that start with "TELLU_PCLEAN_AB"   - consequences for every odometer code you will need to recompute the pre-clean once (it will be added back to the telluDB but it will also be in the reduced directory -- this is the "easiest" option but the slowest

2. remove all entries in the master_tellu_DB that start with "TELLU_PCLEAN_AB" where you do not have the preclean file in your reduced directory --  this is harder than option (1) as you will need to check all your reduced directories for each precleaned file - this will be faster than option 1 as you wont need to pre-clean files you have pre-cleaned yourself (that already exist on disk) - OR you could just keep the TELLU_PCLEAN_AB entries where you already have them in your reduced directories (same difference but maybe easier than finding stuff you don't have)

3. using the master_tellu_DB copy the _tellu_pclean_AB.fits file to the correct reduced/{NIGHT_NAME} path - this is harder than option (2) as you need to figure out which reduced directory each precleaned file should be in - however it is the fastest as you don't need to recompute all the preclean files that were given from Montreal and that you processed there.

I will have to fix this better (more permanently) by getting it from the telluDB instead of the reduced directory

**Comment by njcuk9999** [comment]
Ah yes that would be a good way to find objects! Note that it will add them back as part of creating them - so don't be worried when they come back. When they come back they will also be in your reduced directories so we shouldn't have the same problem after that!

**Comment by pfouque58** [comment]
It seems I may have a similar issue but only for the RV standards when running obj_fit_tellu_db:

01:09:51.038-@|FTELLDB[00413]|W[00-001-00043]: Internal error in recipe obj_fit_tellu_spirou.py. Controller recipe (obj_fit_tellu_db_spirou) cannot continue
01:09:51.170- |FTELLDB[00413]|
01:09:51.245- |FTELLDB[00413]| ***************************************************************************
01:09:51.320- |FTELLDB[00413]|Errors:
01:09:51.396- |FTELLDB[00413]| ***************************************************************************
01:09:51.471- |FTELLDB[00413]|
01:09:51.539- |FTELLDB[00413]|
01:09:51.607- |FTELLDB[00413]| ***************************************************************************
01:09:51.673-@|FTELLDB[00413]|W[40-503-00019]: Error found for ID='DBFTELLU_69'
01:09:51.740-@|FTELLDB[00413]|	obj_fit_tellu_spirou.py 20AQ02-Feb04 2468643o_pp_e2dsff_AB.fits --program=DBFTELLU
01:09:51.810- |FTELLDB[00413]| ***************************************************************************
01:09:51.878- |FTELLDB[00413]|
 E[01-001-00012]: Read Error: File '2468643o_pp_tellu_pclean_AB.fits' could not
 	be found in directory '/data/spirou1/apero-data/offline/reduced/20AQ02-Feb04'.
 	 function = io.drs_fits.py.readfits()
01:09:51.954- |FTELLDB[00413]|
01:09:52.022- |FTELLDB[00413]|
01:09:52.138- |FTELLDB[00413]| ***************************************************************************
01:09:52.226- |FTELLDB[00413]|
01:09:52.314- |FTELLDB[00413]| ***************************************************************************
01:09:52.404-@|FTELLDB[00413]|W[40-503-00019]: Error found for ID='DBFTELLU_70'
01:09:52.491-@|FTELLDB[00413]|	obj_fit_tellu_spirou.py 20AQ02-Feb04 2468644o_pp_e2dsff_AB.fits --program=DBFTELLU
01:09:52.580- |FTELLDB[00413]| ***************************************************************************
01:09:52.667- |FTELLDB[00413]|
 E[01-001-00012]: Read Error: File '2468644o_pp_tellu_pclean_AB.fits' could not
 	be found in directory '/data/spirou1/apero-data/offline/reduced/20AQ02-Feb04'.
 	 function = io.drs_fits.py.readfits()
01:09:52.746- |FTELLDB[00413]|

…

01:09:56.906- |FTELLDB[00413]|
01:09:56.973- |FTELLDB[00413]|
01:09:57.045- |FTELLDB[00413]| ***************************************************************************
01:09:57.114- |FTELLDB[00413]|
01:09:57.192- |FTELLDB[00413]| ***************************************************************************
01:09:57.270-@|FTELLDB[00413]|W[40-503-00019]: Error found for ID='DBMKTEMP_18'
01:09:57.341-@|FTELLDB[00413]|	obj_mk_template_spirou.py GL436 --program=DBMKTEMP
01:09:57.409- |FTELLDB[00413]| ***************************************************************************
01:09:57.478- |FTELLDB[00413]|
 E[01-010-00001]: Unhandled error has occurred: Error <class 'IndexError'> 
 
 Traceback (most recent call last):
  File "/data/spirou/apero/apero-drs/apero/core/core/drs_startup.py", line 355, in run
    llmain = func(recipe, params)
  File "/data/spirou/apero/apero-drs/apero/recipes/spirou/obj_mk_template_spirou.py", line 215, in __main__
    s1d_props = telluric.make_1d_template_cube(params, recipe, *margs)
  File "/data/spirou/apero/apero-drs/apero/science/telluric/template_tellu.py", line 320, in make_1d_template_cube
    reffile.set_filename(filenames[0])
 IndexError: index 0 is out of bounds for axis 0 with size 0
 
01:09:57.586- |FTELLDB[00413]|
01:09:57.656- |FTELLDB[00413]|
01:09:57.724- |FTELLDB[00413]| ***************************************************************************
01:09:57.795- |FTELLDB[00413]|
01:09:57.867- |FTELLDB[00413]| ***************************************************************************
01:09:57.934-@|FTELLDB[00413]|W[40-503-00019]: Error found for ID='DBMKTEMP_19'
01:09:58.002-@|FTELLDB[00413]|	obj_mk_template_spirou.py GL514 --program=DBMKTEMP
01:09:58.078- |FTELLDB[00413]| ***************************************************************************
01:09:58.151- |FTELLDB[00413]|
 E[01-010-00001]: Unhandled error has occurred: Error <class 'IndexError'> 
 
 Traceback (most recent call last):
  File "/data/spirou/apero/apero-drs/apero/core/core/drs_startup.py", line 355, in run
    llmain = func(recipe, params)
  File "/data/spirou/apero/apero-drs/apero/recipes/spirou/obj_mk_template_spirou.py", line 215, in __main__
    s1d_props = telluric.make_1d_template_cube(params, recipe, *margs)
  File "/data/spirou/apero/apero-drs/apero/science/telluric/template_tellu.py", line 320, in make_1d_template_cube
    reffile.set_filename(filenames[0])
 IndexError: index 0 is out of bounds for axis 0 with size 0
 
01:09:58.297- |FTELLDB[00413]|
01:09:58.384- |FTELLDB[00413]|
01:09:58.472- |FTELLDB[00413]| ***************************************************************************
01:09:58.560- |FTELLDB[00413]|
01:09:59.288- |FTELLDB[00413]|A[40-004-00001]: No outputs to index, skipping indexing
01:09:59.380-*|FTELLDB[00413]| ***************************************************************************
01:09:59.468-*|FTELLDB[00413]|I[40-003-00001]: Recipe obj_fit_tellu_db_spirou has been successfully completed
01:09:59.556-*|FTELLDB[00413]| ***************************************************************************
(venv) spdrs@apiki:/data/spirou1/apero-data/offline/msg/processing/APEROG-PID-00016026403582599980-4MLE_apero_processing_group$

**Comment by njcuk9999** [comment]
Well if Andres is getting it you would get it too (as you are both using my telluric database) - please follow one of the quick fixes above!

**Comment by pfouque58** [comment]
I used option 3 (copy the missing pclean files to the reduced directory) and reprocessed using the mini_run.ini rather than the complete_run.ini and it works. The only remaining point is to know why the pclean files were missing for the telluric and RV standards in the reduced directory while they were present for all the other object files

**Comment by njcuk9999** [comment]
This is because they were present in the telluric database from Montreal.
i.e. we processed this odometer code with mk_tellu or fit_tellu 

This means your telluric database (from Montreal) will have a PCLEAN entry - but the bug in the code was that it looked in your reduced directory not your telluric database directory (and obviously you don't have the reduced directory from Montreal).

This is corrected in version 0.6.132

**Comment by pfouque58** [comment]
OK got it. Thanks
